src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/config/DeleteTaskFeatureToggleInterceptor.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static uk.gov.hmcts.reform.wataskmanagementapi.config.features.FeatureFlag.WA_DELETE_TASK_BY_CASE_ID;
+
+public class DeleteTaskFeatureToggleInterceptor implements HandlerInterceptor {
+
+    @Autowired
+    private LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider;
+
+    @Override
+    public boolean preHandle(final HttpServletRequest request,
+                             final HttpServletResponse response,
+                             final Object handler) {
+
+        return launchDarklyFeatureFlagProvider.getBooleanValue(
```

Comment:
What would be the http response code if returning `false` (guessing there's a default it uses)?  I think setting it to something like a 501 or 503 would be good.